From d385caeac4e9632bedb9ea0bb812beb4b03b7e92 Mon Sep 17 00:00:00 2001
From: "Chen, Gang G" <gang.g.chen@intel.com>
Date: Mon, 13 May 2024 14:31:04 +0800
Subject: [PATCH 52/54] Removed Verity Support.

Removing drm-verity support as it no longer
supported by google.
Original Jira - OAM-106853

Tracked-On: OAM-118547
Signed-off-by: Tanuj Tekriwal <tanuj.tekriwal@intel.com>
Signed-off-by: Chen, Gang G <gang.g.chen@intel.com>
---
 Android.mk | 75 ++----------------------------------------------------
 1 file changed, 2 insertions(+), 73 deletions(-)

diff --git a/Android.mk b/Android.mk
index fab7fcc..a832373 100644
--- a/Android.mk
+++ b/Android.mk
@@ -162,40 +162,6 @@ LOCAL_MODULE := kernelflinger-$(TARGET_BUILD_VARIANT)
 
 
 # if dm-verity is disabled for eng purpose skip the oem-cert
-ifeq ($(PRODUCT_SUPPORTS_VERITY), true)
-kf_intermediates := $(call intermediates-dir-for,EFI,kernelflinger)
-
-VERITY_CERT := $(kf_intermediates)/verity.cer
-PADDED_VERITY_CERT := $(kf_intermediates)/verity.padded.cer
-OEMCERT_OBJ := $(kf_intermediates)/oemcert.o
-
-$(VERITY_CERT): $(PRODUCTS.$(INTERNAL_PRODUCT).PRODUCT_VERITY_SIGNING_KEY).x509.pem
-	$(transform-pem-cert-to-der-cert)
-
-$(PADDED_VERITY_CERT): $(VERITY_CERT)
-	$(call pad-binary, 4096)
-
-ifeq ($(TARGET_UEFI_ARCH),x86_64)
-    ELF_OUTPUT := elf64-x86-64
-else
-    ELF_OUTPUT := elf32-i386
-endif
-
-sym_binary := $(shell echo _binary_$(PADDED_VERITY_CERT) | sed "s/[\/\.-]/_/g")
-$(OEMCERT_OBJ): $(PADDED_VERITY_CERT)
-	mkdir -p $(@D) && \
-	$(EFI_OBJCOPY) --input binary --output $(ELF_OUTPUT) --binary-architecture i386 $< $@ && \
-	$(EFI_OBJCOPY) --redefine-sym $(sym_binary)_start=_binary_oemcert_start \
-                       --redefine-sym $(sym_binary)_end=_binary_oemcert_end \
-                       --redefine-sym $(sym_binary)_size=_binary_oemcert_size \
-                       --rename-section .data=.oemkeys $@ $@
-
-LOCAL_GENERATED_SOURCES := $(OEMCERT_OBJ)
-else # PRODUCT_SUPPORTS_VERITY
-ifneq (,$(filter user userdebug, $(TARGET_BUILD_VARIANT)))
-
-endif
-endif # PRODUCT_SUPPORTS_VERITY
 
 kf_intermediates := $(call intermediates-dir-for,EFI,kernelflingeravb)
 
@@ -209,7 +175,7 @@ BOOTLOADER_AVB_KEY_PATH := $(BOARD_AVB_KEY_PATH)
 endif
 
 $(AVB_PK): $(BOOTLOADER_AVB_KEY_PATH)
-	external/avb/avbtool extract_public_key --key $< --output $@
+	external/avb/avbtool.py extract_public_key --key $< --output $@
 
 $(PADDED_AVB_PK): $(AVB_PK)
 	$(call pad-binary, 4096)
@@ -262,9 +228,6 @@ endif
 
 LOCAL_CFLAGS := $(SHARED_CFLAGS)
 
-ifeq ($(PRODUCT_SUPPORTS_VERITY), true)
-LOCAL_OBJCOPY_FLAGS := -j .oemkeys
-endif
 
 LOCAL_OBJCOPY_FLAGS := -j .oemkeys
 
@@ -314,7 +277,7 @@ BOOTLOADER_AVB_KEY_PATH := $(BOARD_AVB_KEY_PATH)
 endif
 
 $(KFINS_AVB_PK): $(BOOTLOADER_AVB_KEY_PATH)
-	external/avb/avbtool extract_public_key --key $< --output $@
+	external/avb/avbtool.py extract_public_key --key $< --output $@
 
 $(KFINS_PADDED_AVB_PK): $(KFINS_AVB_PK)
 	$(call pad-binary, 4096)
@@ -411,36 +374,6 @@ ifneq ($(strip $(KERNELFLINGER_USE_UI)),false)
     LOCAL_SRC_FILES += \
         ux.c
 endif
-ifeq ($(PRODUCT_SUPPORTS_VERITY),true)
-keys4abl_intermediates := $(call intermediates-dir-for,ABL,keys)
-
-ABL_VERITY_CERT := $(keys4abl_intermediates)/verity.cer
-ABL_PADDED_VERITY_CERT := $(keys4abl_intermediates)/verity.padded.cer
-ABL_OEMCERT_OBJ := $(keys4abl_intermediates)/oemcert.o
-
-$(ABL_VERITY_CERT): $(PRODUCTS.$(INTERNAL_PRODUCT).PRODUCT_VERITY_SIGNING_KEY).x509.pem
-	$(transform-pem-cert-to-der-cert)
-
-$(ABL_PADDED_VERITY_CERT): $(ABL_VERITY_CERT)
-	$(call pad-binary, 4096)
-
-ifeq ($(TARGET_IAFW_ARCH),x86_64)
-    ELF_OUTPUT := elf64-x86-64
-else
-    ELF_OUTPUT := elf32-i386
-endif
-
-abl_sym_binary := $(shell echo _binary_$(ABL_PADDED_VERITY_CERT) | sed "s/[\/\.-]/_/g")
-$(ABL_OEMCERT_OBJ): $(ABL_PADDED_VERITY_CERT)
-	mkdir -p $(@D) && \
-	$(EFI_OBJCOPY) --input binary --output $(ELF_OUTPUT) --binary-architecture i386 $< $@ && \
-	$(EFI_OBJCOPY) --redefine-sym $(abl_sym_binary)_start=_binary_oemcert_start \
-                       --redefine-sym $(abl_sym_binary)_end=_binary_oemcert_end \
-                       --redefine-sym $(abl_sym_binary)_size=_binary_oemcert_size \
-                       --rename-section .data=.oemkeys $@ $@
-
-LOCAL_GENERATED_SOURCES := $(ABL_OEMCERT_OBJ)
-endif #.PRODUCT_SUPPORTS_VERITY == true
 
 keys4abl_intermediates := $(call intermediates-dir-for,ABL,keys4abl)
 
@@ -525,10 +458,6 @@ ifneq ($(TARGET_BUILD_VARIANT),user)
     LOCAL_SRC_FILES += unittest.c
 endif
 
-ifeq ($(PRODUCT_SUPPORTS_VERITY),true)
-LOCAL_GENERATED_SOURCES := $(ABL_OEMCERT_OBJ)
-endif
-
 LOCAL_GENERATED_SOURCES += $(ABL_AVB_PK_OBJ)
 LOCAL_C_INCLUDES := \
 	$(addprefix $(LOCAL_PATH)/,avb)
-- 
2.25.1

